### PR TITLE
chore(clouddriver/oracle): Get the Oracle API jar from Maven.

### PR DESCRIPTION
**Pull request by plumpy**
cc @ezimanyi @shihchang @guoyongzhang

### CODE REVIEW

**Comment by emjburns** [comment]
🎊

**Comment by andrewbackes** [comment]
Starting with this commit -
```
commit 6a1be53b30894a293fe8d4a181dd1862340b4617 (HEAD)
Author: Michael Plump <plumpy@google.com>

    chore(clouddriver/oracle): Get the Oracle API jar from Maven. (#3695)
```

The `clouddriver-lambda` module throws an exception for me when caching lambdas -
```
2019-05-30 12:15:55.408  WARN 7189 --- [cutionAction-10] c.n.s.c.cache.LoggingInstrumentation     : com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider:aws/us-west-2/LambdaCachingAgent completed with one or more failures

com.amazonaws.services.lambda.model.AWSLambdaException: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details. (Service: AWSLambda; Status Code: 403; Error Code: InvalidSignatureException; Request ID: 59657453-830f-11e9-8ffe-bdcf36568289)
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1712) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1367) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512) ~[aws-java-sdk-core-1.11.534.jar:na]
        at com.amazonaws.services.lambda.AWSLambdaClient.doInvoke(AWSLambdaClient.java:3139) ~[aws-java-sdk-lambda-1.11.534.jar:na]
        at com.amazonaws.services.lambda.AWSLambdaClient.invoke(AWSLambdaClient.java:3106) ~[aws-java-sdk-lambda-1.11.534.jar:na]
        at com.amazonaws.services.lambda.AWSLambdaClient.invoke(AWSLambdaClient.java:3095) ~[aws-java-sdk-lambda-1.11.534.jar:na]
        at com.amazonaws.services.lambda.AWSLambdaClient.executeListVersionsByFunction(AWSLambdaClient.java:2275) ~[aws-java-sdk-lambda-1.11.534.jar:na]
        at com.amazonaws.services.lambda.AWSLambdaClient.listVersionsByFunction(AWSLambdaClient.java:2246) ~[aws-java-sdk-lambda-1.11.534.jar:na]
        at com.netflix.spinnaker.clouddriver.lambda.provider.agent.LambdaCachingAgent.listFunctionRevisions(LambdaCachingAgent.java:156) ~[clouddriver-lambda.jar:na]
        at com.netflix.spinnaker.clouddriver.lambda.provider.agent.LambdaCachingAgent.loadData(LambdaCachingAgent.java:127) ~[clouddriver-lambda.jar:na]
        at com.netflix.spinnaker.cats.agent.CachingAgent$CacheExecution.executeAgentWithoutStore(CachingAgent.java:85) ~[cats-core.jar:na]
        at com.netflix.spinnaker.cats.agent.CachingAgent$CacheExecution.executeAgent(CachingAgent.java:75) ~[cats-core.jar:na]
        at com.netflix.spinnaker.cats.redis.cluster.ClusteredAgentScheduler$AgentExecutionAction.execute(ClusteredAgentScheduler.java:335) ~[cats-redis.jar:na]
        at com.netflix.spinnaker.cats.redis.cluster.ClusteredAgentScheduler$AgentJob.run(ClusteredAgentScheduler.java:305) ~[cats-redis.jar:na]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_191]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_191]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_191]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_191]
        at java.lang.Thread.run(Thread.java:748) [na:1.8.0_191]
```

The commit right before it - 
```
commit 45009129db70ed4a15c6a72d96010ec013a4a646
Author: Eric Zimanyi <ezimanyi@google.com>

    fix(kubernetes): Fix compiler warnings (#3698)
```
does not throw the exception.

Looking at this PR, I'm not exactly sure why it would cause the problem though. Are any of you seeing this issue?

File: clouddriver-oracle/clouddriver-oracle.gradle [hunk 1]
```diff
@@ -1,68 +1,8 @@
...
-  exclude "**/*guava*.jar"
-  exclude "**/commons*.jar"
-  exclude "**/aopalliance*.jar"
-  exclude "**/javassist*.jar"
-  exclude "**/slf*.jar"
-  exclude "**/osgi*.jar"
-  exclude "**/validation*.jar"
-  exclude "**/jsr305*.jar"
-  exclude "**/json-smart*.jar"
-  exclude "**/oci-java-sdk-full-shaded-*.jar"
```
Comment (robzienert) [inline clouddriver-oracle/clouddriver-oracle.gradle]: Looks like we need to double back and continue excluding some of these deps from being transitively included by the new artifact.

**Comment by plumpy** [comment]
Here's a diff of the distribution jars before/after this commit. I don't see anything obvious that might be causing this :/

bcprov-jdk15on is from bouncycastle, that's the only thing that's really related...

Only in new: aopalliance-repackaged-2.5.0-b42.jar
Only in old: bcprov-jdk15on-1.60.jar
Only in old: checker-compat-qual-2.0.0.jar
Binary files old/clouddriver-oracle.jar and new/clouddriver-oracle.jar differ
Only in new: hk2-api-2.5.0-b42.jar
Only in new: hk2-locator-2.5.0-b42.jar
Only in new: hk2-utils-2.5.0-b42.jar
Only in old: httpclient-4.5.jar
Only in old: httpcore-4.4.1.jar
Only in old: javassist-3.21.0-GA.jar
Only in new: javassist-3.22.0-CR2.jar
Only in old: javax.annotation-api-1.2.jar
Only in old: javax.inject-2.5.0-b05.jar
Only in new: javax.inject-2.5.0-b42.jar
Only in old: javax.ws.rs-api-2.0.1.jar
Only in new: javax.ws.rs-api-2.1.jar
Only in old: jcip-annotations-1.0.jar
Only in new: jersey-client-2.27.jar
Only in new: jersey-common-2.27.jar
Only in new: jersey-entity-filtering-2.27.jar
Only in new: jersey-hk2-2.27.jar
Only in new: jersey-media-json-jackson-2.27.jar
Only in old: nimbus-jose-jwt-4.9.jar
Only in old: oci-java-sdk-addons-apache-1.3.2.jar
Only in new: oci-java-sdk-common-1.5.2.jar
Only in new: oci-java-sdk-core-1.5.2.jar
Only in old: oci-java-sdk-full-1.3.2.jar
Only in new: oci-java-sdk-identity-1.5.2.jar
Only in new: oci-java-sdk-loadbalancer-1.5.2.jar
Only in new: oci-java-sdk-objectstorage-1.5.2.jar
Only in new: oci-java-sdk-objectstorage-extensions-1.5.2.jar
Only in new: oci-java-sdk-objectstorage-generated-1.5.2.jar
Only in new: osgi-resource-locator-1.0.1.jar

**Comment by plumpy** [comment]
Any chance you could try after #3743 and see if that fixes it? It's my best idea, but it may not be a very good idea.

**Comment by andrewbackes** [comment]
I tried it on 
```
commit 60ee3441c23f5608383f3b5a0cdfe424097357db (tag: v5.7.0)
Author: spinnakerbot <spinbot@spinnaker.io>
Date:   Mon Jun 3 18:33:06 2019 -0400

    chore(dependencies): Autobump korkVersion (#3743)
```
and the problem is still there.

Just to see if things if it would fix it, I did a
```
$git revert 6a1be53b30894a293fe8d4a181dd1862340b4617
```
and the Lambda caching problem went away. However, I didn't run extensive integration tests on all providers. Would it be reasonable to temporarily revert this until we get back into a working state?

**Comment by plumpy** [comment]
Could you open an issue so we can discuss it there? This doesn't feel like the right place...

**Comment by andrewbackes** [comment]
https://github.com/spinnaker/spinnaker/issues/4484

**Pull request merged**